core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -173,28 +173,28 @@ case class PSBT(
     require(
       index < inputMaps.size,
       s"index must be less than the number of input maps present in the psbt, $index >= ${inputMaps.size}")
-    require(!inputMaps(index).isFinalized,
+
+    val map = inputMaps(index)
+    require(!map.isFinalized,
             s"Cannot update an InputPSBTMap that is finalized, index: $index")
 
-    val previousElements = inputMaps(index).elements
     val txIn = transaction.inputs(index)
     val elements =
       if (txIn.previousOutput.vout.toInt < tx.outputs.size) {
         val out = tx.outputs(txIn.previousOutput.vout.toInt)
 
         val outIsWitnessScript =
           WitnessScriptPubKey.isWitnessScriptPubKey(out.scriptPubKey.asm)
-        val hasWitScript = inputMaps(index).witnessScriptOpt.isDefined
+        val hasWitScript = map.witnessScriptOpt.isDefined
 
         if (outIsWitnessScript || hasWitScript) {
-          previousElements.filterNot(_.key.head == WitnessUTXOKeyId.byte) :+ WitnessUTXO(
-            out)
+          map.filterRecords(WitnessUTXOKeyId) :+ WitnessUTXO(out)
         } else {
-          previousElements.filterNot(_.key.head == NonWitnessUTXOKeyId.byte) :+ NonWitnessOrUnknownUTXO(
-            tx)
+          map.filterRecords(NonWitnessUTXOKeyId) :+ NonWitnessOrUnknownUTXO(tx)
         }
       } else {
-        previousElements
+        throw new IllegalArgumentException(
```

Comment:
nit: `IllegalStateException` right?

Reply:
I think `IllegalArgumentException` is correct here because if we reach here it means that the transaction given as an argument had less outputs than it would need to be a valid previous transaction for this input